src/general/disallowanonymous/src_test.rego

```diff
@@ -222,6 +222,18 @@ test_allowed_multiple_role_multiple_subjects_mix_rolebinding {
     count(results) == 0
 }
 
+test_authenticated_with_disallow_authenticated {
+    inp := {"review": rolebinding([{"name": "system:authenticated", "kind": "Group"}, {"name": "system:anonymous", "kind": "User"}], "role-2"), "parameters": { "allowedRoles": [], "disallowAuthenticated": true }}
+    results := violation with input as inp
+    count(results) == 1
+}
+
+test_authenticated_with_disallow_authenticated_false {
```

Comment:
I would add a test with no `"disallowAuthenticated"` parameter at all.

Reply:
There are other tests above that lack the disallowAuthenticated parameter.  I can dedup a bit to make that more obvious.

Reply:
Added test case.

Reply:
To clarify I meant no parameter object at all, but with rebasing on https://github.com/open-policy-agent/gatekeeper-library/pull/578 that should also be covered.